test/standalone/c_compiler/build.zig

```diff
@@ -1,20 +1,25 @@
 const std = @import("std");
-const builtin = @import("builtin");
 const Builder = std.build.Builder;
 const CrossTarget = std.zig.CrossTarget;
 
-// TODO integrate this with the std.build executor API
-fn isRunnableTarget(t: CrossTarget) bool {
-    if (t.isNative()) return true;
+fn isSingleThreadedTarget(t: CrossTarget) bool {
+    // WASM target will be detected automatically as single threaded
 
-    return (t.getOsTag() == builtin.os.tag and
-        t.getCpuArch() == builtin.cpu.arch);
+    // C++ lib doesn't work on ARM on multi-threaded mode yet:
+    // https://reviews.llvm.org/D75183 (abandoned, used as a ref)
+    return t.getCpuArch() == .arm;
```

Comment:
This logic should be moved to Compilation.zig where it deals with the Options struct to affect the choice of `single_threaded`.

If the user passes, for example, `-lc++ -target arm-linux`, then they should get single_threaded false by default. If they do not link libc++ then it should default to multi-threaded (status quo). Finally if the user requests both multi-threaded, libc++, and arm, then they should get an error: `-lc++ -target arm-linux -fno-single-threaded` should cause "error: libc++ does not work on multi-threaded ARM yet".

Reply:
Done, verified by:
```
cd test/standalone/c_compiler
zig build-exe -lc++ -target arm-linux-musleabi test.cpp
zig build-exe -lc++ -target arm-linux-musleabi -fno-single-threaded test.cpp
zig build-exe -lc++ -target arm-linux-musleabi -fsingle-threaded test.cpp
```

Only the latter works, other fail with:
```
warning(compilation): libc++ does not work on multi-threaded ARM yet.
For more details: https://github.com/ziglang/zig/issues/6573
error: unable to create compilation: TargetRequiresSingleThreaded
```